### ISSUE TITLE
Proposal: remove arg server from buf processes

### ISSUE THREAD

**Issue by elgiano** [flucoma/flucoma-sc#139]
I know this is a breaking interface change, but since it's impossible to have a buffer on one server and run the analysis on another, I think the server argument is redundant for all buffer processes. Such processes can get the server instance from the source buffer that gets passed to them:
```
// in, e.g. FluidBufMFCC.process
^this.new(src.server, nil, [features])
// instead of: ^this.new(server, nil,[features])
```
Since BufProcesses have already lots of args usually, I think that removing a redundant one would contribute to cleaner (e.g. more understandable/actionable) code

**Comment by weefuzzy** [comment]
Good point (and I guess we should assert that the `server` for each passed `Buffer` actually matches). The scale of breakage will be pretty immense though, so I'd be inclined to think carefully about how / when such a change could be deployed. It might be more of a v2 thing?
